shared_python/ValidateTags.py

```diff
@@ -0,0 +1,187 @@
+        "relationships": 6,
+        "tags": 7
+    }
+    dict_rating = {
+        "Not Rated": 1,
+        "General Audiences": 2,
+        "Teen And Up Audiences": 3,
+        "Mature": 4,
+        "Explicit": 5
+    }
+    dict_warning = {
+        "Choose Not To Use Archive Warnings": 1,
+        "Graphic Depictions Of Violence": 2,
+        "Major Character Death": 3,
+        "No Archive Warnings Apply": 4,
+        "Rape/Non-Con": 5,
+        "Underage": 6
+    }
+    dict_category = {
+        "Gen": 1,
+        "F/M": 2,
+        "M/M": 3,
+        "F/F": 4,
+        "Multi": 5,
+        "Other": 6
+    }
+
+    ## Functions for accessing dictionaries
+    def identify_tag_type(self, tag_type):
+        return self.dict_tag_type.get(tag_type, 0)
+
+    def identify_rating(self, tag):
+        return self.dict_rating.get(tag, 0)
+
+    def identify_warning(self, tag):
+        return self.dict_warning.get(tag, 0)
+
+    def identify_category(self, tag):
+        return self.dict_category.get(tag, 0)
+
+    ## Print statements
+    def print_tag_correction(self, before, after):
+        print('\r\033[96m', end='')
+        self.log.info('Correction successful. "' + before + '" is now "' + after+ '"')
+        print('\x1b[0m', end='')
+
+    def print_tag_warning(self, tag, isType):
+        print('\r\033[93m', end='')
+        if (isType):
+            self.log.warning('Warning: "' + tag + '" is not a valid TAG TYPE. Attempting self correction...')
+        else:
+            self.log.warning('Warning: "' + tag + '" is not a valid TAG. Attempting self correction...')
+        print('\x1b[0m', end='')
+
+    def print_fail_self(self):
+        print('\r\033[91m', end='')
+        self.log.warning('All attempts at self correction have failed. Manual correction required.')
+        print('\x1b[0m', end='')
+
+    def print_fail(self, tag_name):
+        print('\r\033[91m', end='')
+        if (tag_name):
+            self.log.warning('Manual Input Failed. "' + tag_name + '" has failed re-check.')
+        else:
+            self.log.warning('Manual Input Failed. This field cannot be empty.')
+        print('\x1b[0m', end='')
+
+    def prompt_correction(self, tag_name, tag_type, isType):
+        print ('\rThe following values are accepted:')
+        if (isType):
+            print('\r' + self.list_dicts(None, tag_type))
+        else:
+            print('\r' + self.list_dicts(tag_name, tag_type))
+        prompt = '\rPlease enter the correct name for "' + tag_name + '"'
+        if (isType):
+            prompt += ' (Press ENTER to default to "tags"): '
+        else:
+            prompt += ': '
+        tag_correction = input(prompt)
+        if (isType and not tag_correction):
+            tag_correction = "tags"
+        return tag_correction
+
+    ## Self Correction Logic
+    ## NOTE: This is for very light corrections only.
+    def correct_tag_type(self, tag_type):
+        # Attempt self correciton by making whole word lowercase
```

Comment:
typo: `correction`

Reply:
thanks for catching this one. typing is hard.